src/app.ts

```diff
@@ -13,14 +13,38 @@ interface TemperatureSummary {
   average: number
 }
 
+const dataByCity: { [key: string]: TemperatureReading[] } = {}
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((reading: TemperatureReading): void => {
+    if (!dataByCity[reading.city]) {
+      dataByCity[reading.city] = []
+    }
+    dataByCity[reading.city].push(reading)
+  })
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const readings: TemperatureReading[] = dataByCity[city]?.filter(
+    (reading) => reading.time.toDateString() === date.toDateString(),
+  )
+  if (!readings || readings.length === 0) {
+    return null
+  }
+
+  const temperatures: number[] = readings.map(
+    (reading: TemperatureReading) => reading.temperature,
+  )
+  const first: number = temperatures[0]
+  const last: number = temperatures[temperatures.length - 1]
+  const high: number = Math.max(...temperatures)
+  const low: number = Math.min(...temperatures)
+  const average: number =
+    temperatures.reduce((sum: number, temp: number) => sum + temp) /
+    temperatures.length
```

Comment:
The less types we write, the better, we can let TypeScript to infer the types in this case.

```suggestion
  const temperatures = readings.map(
    (reading: TemperatureReading) => reading.temperature,
  )
  const first = temperatures[0]
  const last = temperatures[temperatures.length - 1]
  const high = Math.max(...temperatures)
  const low = Math.min(...temperatures)
  const average =
    temperatures.reduce((sum: number, temp: number) => sum + temp) /
    temperatures.length
```